src/NftReward.sol

```diff
@@ -49,22 +50,34 @@ contract NftReward is ERC721, Ownable, Pausable, EIP712 {
     }
 
     /**
-     * @notice Contract constructor
+     * @notice _disableInitializers in the constructor, 
+     * this prevents initialization of the implementation contract itself, 
+     * as extra protection to prevent an attacker from initializing it.
+     */
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Contract initializer (replaces constructor)
      * @param _tokenName NFT token name
      * @param _tokenSymbol NFT token symbol
      * @param _initialOwner Initial owner name
      * @param _minter Minter address
      */
-    constructor (
+    function initialize(
         string memory _tokenName, 
         string memory _tokenSymbol,
         address _initialOwner,
         address _minter
     ) 
-        ERC721(_tokenName, _tokenSymbol) 
-        Ownable(_initialOwner)
-        EIP712("NftReward-Domain", "1")
+        public 
+        initializer 
     {
+        __ERC721_init(_tokenName, _tokenSymbol);
+        __Ownable_init(_initialOwner);
+        __EIP712_init("NftReward-Domain", "1");
+        __UUPSUpgradeable_init();
```

Comment:
Don't we need to call https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/5dd8d5960d7a7a00271a38716807ff420747c602/contracts/utils/PausableUpgradeable.sol#L56 here?